src/main/java/bio/terra/service/filedata/FileService.java

```diff
@@ -177,6 +183,71 @@ public String ingestBulkFileArray(
         .submit();
   }
 
+  public List<FileModel> listDatasetFiles(String datasetId, Integer offset, Integer limit) {
+    Dataset dataset = datasetService.retrieve(UUID.fromString(datasetId));
+    CloudPlatformWrapper cloudPlatformWrapper = CloudPlatformWrapper.of(dataset.getCloudPlatform());
+    String collectionId = datasetId;
+    List<FireStoreFile> results;
+    if (cloudPlatformWrapper.isGcp()) {
+      try {
+        collectionId = String.format("%s-files", datasetId);
+        results = fileDao.retrieveFiles(dataset, collectionId, offset, limit);
+      } catch (InterruptedException | ExecutionException ex) {
+        throw new FileSystemExecutionException(
+            "Unexpected interruption during file system processing", ex);
+      }
+    } else {
+      collectionId = FILES_TABLE.toTableName(UUID.fromString(collectionId));
```

Comment:
I think `UUID.fromString(collectionId)` is `dataset.getId()`
```suggestion
      collectionId = FILES_TABLE.toTableName(dataset.getId());
```